DiscordIan/Module/OpenWeather.cs

```diff
@@ -0,0 +1,317 @@
+        [Alias("weather", "w")]
+        public async Task CurrentAsync([Remainder]
+            [Summary("The address or location for current weather conditions")] string location)
+        {
+            string coords = await _cache.GetStringAsync(location);
+            string locale = coords == null ? null : await _cache.GetStringAsync(coords);
+
+            if (string.IsNullOrEmpty(coords) && string.IsNullOrEmpty(location))
+            {
+                await ReplyAsync("Please provide a location.");
+                return;
+            }
+            Model.Geocodio.Location coordinates = null;
+            Model.Geocodio.AddressComponents address = null;
+            if (coords == null || locale == null)
+            {
+                try
+                {
+                    (coordinates, address) = await GeocodeAddressAsync(location);
+
+                    if (coordinates != null && address != null)
+                    {
+                        locale = address.City + ", " + address.State;
+                        string latlong = String.Format("{0},{1}", coordinates.Lat.ToString(), coordinates.Lng.ToString());
+
+                        await _cache.SetStringAsync(location, latlong,
+                            new DistributedCacheEntryOptions { AbsoluteExpirationRelativeToNow = TimeSpan.FromHours(4) });
+
+                        await _cache.SetStringAsync(latlong, locale,
+                            new DistributedCacheEntryOptions { AbsoluteExpirationRelativeToNow = TimeSpan.FromHours(4) });
+                    }
+                }
+                catch (Exception ex)
+                {
+                    //Geocode failed, revert to OpenWeatherMap's built-in resolve.
+                }
+            }
+            else
+            {
+                coordinates = new Model.Geocodio.Location() { Lat = double.Parse(coords.Split(",")[0]), Lng = double.Parse(coords.Split(",")[1]) };
+            }
+
+            string message;
+            Discord.Embed embed;
+
+            if (coordinates == null || locale == null)
+                (message, embed) = await GetWeatherResultAsync(location);
+            else
+                (message, embed) = await GetWeatherResultAsync(coordinates, locale);
+
+
+            if (message == null)
+            {
+                await ReplyAsync("No weather found, sorry!");
+            }
+            else
+            {
+                await ReplyAsync(message, false, embed);
+            }
+        }
+
+        private async Task<(string, Discord.Embed)>
+            GetWeatherResultAsync(Model.Geocodio.Location coordinates, string location)
+        {
+            var headers = new Dictionary<string, string>
+            {
+                { "User-Agent", "DiscorIan Discord bot" }
+            };
+
+            var uriCurr = new Uri(string.Format(_options.IanOpenWeatherMapEndpointCoords,
+                HttpUtility.UrlEncode($"{coordinates.Lat}"),
```

Comment:
This should work as `HttpUtility.UrlEncode(coordinates.Lat)` or `HttpUtility.UrlEncode(coordinates.Lat.ToString())`.